pandas/tests/io/formats/test_format.py

```diff
@@ -3240,13 +3240,22 @@ def test_repr_html_ipython_config(ip):
 
 
 @pytest.mark.parametrize("method", ["to_string", "to_html", "to_latex"])
+@pytest.mark.parametrize("encoding", [None, "utf-8", "gbk", "foo"])
 def test_filepath_or_buffer_arg(
-    float_frame, method, filepath_or_buffer, assert_filepath_or_buffer_equals
+    float_frame,
+    method,
+    filepath_or_buffer,
+    assert_filepath_or_buffer_equals,
+    encoding,
+    filepath_or_buffer_id,
 ):
+    if encoding is not None:
+        assert filepath_or_buffer_id is not None
+        assert filepath_or_buffer_id in ["string", "pathlike"]
```

Comment:
can you use `with pytest.raises` instead.

Reply:
Sure